test/cppsuite/tests/search_near_01.cxx

```diff
@@ -0,0 +1,249 @@
+        testutil_check(cursor->reset(cursor.get()));
+        return valuep;
+    }
+
+    void
+    populate_worker(thread_context *tc)
+    {
+        logger::log_msg(LOG_INFO, "Populate with thread id: " + std::to_string(tc->id));
+        std::string prefix_key;
+        int cmpp;
+        uint64_t collections_per_thread = tc->collection_count;
+        for (int64_t i = 0; i < collections_per_thread; ++i) {
+            collection &coll = tc->db.get_collection(i);
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
+            for (uint64_t j = 0; j < ALPHABET_SIZE; ++j) {
+                for (uint64_t k = 0; k < ALPHABET_SIZE; ++k) {
+                    for (uint64_t count = 0; count < tc->key_count; ++count) {
+                        tc->transaction.begin();
+                        /*
+                         * Generate prefix key of aaa -> zzz, and append a random generated key
+                         * string based on the key size configuration.
+                         */
+                        prefix_key = {alphabet.at(tc->id), alphabet.at(j), alphabet.at(k)};
+                        prefix_key += random_generator::instance().generate_string(tc->key_size);
+                        if (tc->insert(cursor, coll.id, prefix_key)) {
+                            /* We failed to insert, rollback our transaction and retry. */
+                            tc->transaction.rollback();
+                            --count;
+                            continue;
+                        }
+                        /* Commit txn at commit timestamp 100. */
+                        tc->transaction.commit("commit_timestamp=" + tc->tsm->decimal_to_hex(100));
+                    }
+                }
+            }
+        }
+    }
+
+    public:
+    search_near_01(const test_harness::test_args &args) : test(args) {}
+
+    void
+    populate(test_harness::database &database, test_harness::timestamp_manager *tsm,
+      test_harness::configuration *config, test_harness::workload_tracking *tracking) override final
+    {
+        int64_t collection_count, key_count, key_size;
+        std::vector<thread_context *> workers;
+        thread_manager tm;
+
+        /* Validate our config. */
+        collection_count = config->get_int(COLLECTION_COUNT);
+        key_count = keys_per_prefix = config->get_int(KEY_COUNT_PER_COLLECTION);
+        key_size = config->get_int(KEY_SIZE);
+        testutil_assert(key_size > 0);
+        /* Keys must be unique. */
+        testutil_assert(key_count <= pow(10, key_size));
+
+        logger::log_msg(LOG_INFO,
+          "Populate configuration with key size: " + std::to_string(key_size) +
+            " key count: " + std::to_string(key_count) +
+            " number of collections: " + std::to_string(collection_count));
+
+        /* Create n collections as per the configuration. */
+        for (int64_t i = 0; i < collection_count; ++i)
```

Comment:
You are using `uint64_t` for your loop indices at other places, you can use `uint64_t` here too.